### PR TITLE
@broskoski => Don't reference end_at for live sales

### PR DESCRIPTION
**Pull request by sweir27**
This PR has some big refactors, so will have to be heavily tested in staging before it's deployed.

The main goal is to be able to have live sales with no `end_at` set. Mostly this PR is getting rid of places where the `end_at` is assumed, and changing messaging from "Auction closes in: ..." to "Live Bidding Now Open".

Some example screenshots below of the affected pages:

### /auctions
image

### /auction
image

### artwork page banner
(remove clock next to "explore auction")
image

### artist page banner
image

### home page featured auction
image

### CODE REVIEW

**Comment by katarinabatina** [comment]
@sweir27 looking awesome glad to have these changes in!

**Comment by sweir27** [comment]
@broskoski this is 🍏 if you want to take a look :)

File: apps/artist/view_helpers.coffee [hunk 1]
```diff
@@ -103,8 +103,13 @@ module.exports =
...
 
-  formatAuctionDetail: (end) ->
-    minuteFormat = if end.minute() > 0 then ':mm' else ''
-    end.format('[Auction Closes] MMM D [at] h' + minuteFormat + ' A');
+  formatAuctionDetail: (auction) ->
+    if auction.live_start_at and moment().isAfter(auction.live_start_at)
+      'Live Bidding Now Open'
+    else
+      label = if auction.live_start_at then 'Live Bidding Opens' else 'Auction Closes'
+      end = if auction.live_start_at then moment(auction.live_start_at) else moment(auction.end_at)
```
Comment (broskoski) [inline apps/artist/view_helpers.coffee]: What is the point here of setting `end` to `live_start_at`?

File: apps/auction/templates/my_active_bids.jade [hunk 2]
```diff
@@ -22,7 +22,7 @@ if myActiveBids && myActiveBids.length
             td.auction-mab-current-bid
               strong Current Bid: <em>#{bid.sale_artwork.highest_bid.display}</em>
             td.auction-mab-bid-num (#{bid.sale_artwork.counts.bidder_positions} Bids)
-            if ViewHelpers.mpLiveSaleIsOpen(bid.sale_artwork.sale)
+            if bid.sale_artwork.sale.is_live_open
```
Comment (broskoski) [inline apps/auction/templates/my_active_bids.jade]: Phew, much better.

File: apps/home/client/setup_home_page_modules.coffee [hunk 3]
```diff
@@ -18,7 +18,7 @@ contexts =
   popular_artists: (module) ->
     popularArtistsTemplate artists: module.context.artists
   live_auctions: (module) ->
-    auctionTemplate auction: module.context
+    auctionTemplate auctionTimeLabel: auctionTimeLabel(module.context), auctionClosesLabel: auctionClosesLabel(module.context)
```
Comment (broskoski) [inline apps/home/client/setup_home_page_modules.coffee]: Innnnterestinggg. Took me a second to parse this! That may be due to my weird formatting to begin with.

Maybe better as:
```coffeescript
auctionTemplate 
   auctionTimeLabel: auctionTimeLabel(module.context)
   auctionClosesLabel: auctionClosesLabel(module.context)
```

File: apps/home/client/setup_home_page_modules.coffee [hunk 4]
```diff
@@ -37,6 +37,12 @@ setupActiveBidsView = (module, $el, user) ->
 
   mabView.fetch().then -> mabView.render().poll()
 
+auctionTimeLabel = (auction) ->
+  if auction.live_start_at then "Live bidding starts #{moment(auction.live_start_at).format("MMM D")}" else "#{auction.start_at} - #{moment(auction.end_at).format("MMM D")}"
```
Comment (broskoski) [inline apps/home/client/setup_home_page_modules.coffee]: Same here, maybe easier to read as multiple lines?

File: models/sale.coffee [hunk 5]
```diff
@@ -118,7 +101,7 @@ module.exports = class Sale extends Backbone.Model
     @isAuction() and moment().isAfter(@get 'registration_ends_at')
 
   isLiveAuction: ->
-    @get('live_start_at') and moment(@get('live_start_at')).isBefore(moment(@get('end_at')))
+    @get('live_start_at')
```
Comment (broskoski) [inline models/sale.coffee]: Nice

File: test/models/mixins/clock.coffee [hunk 6]
```diff
@@ -9,7 +9,7 @@ sd = require('sharify').data
 class Model extends Backbone.Model
   _.extend @prototype, ClockMixin
 
-describe 'Image Sizes Mixin', ->
+describe 'Clock Mixin', ->
```
Comment (broskoski) [inline test/models/mixins/clock.coffee]: :)

File: apps/artist/view_helpers.coffee (hunk 1 above)
Comment (sweir27) [inline apps/artist/view_helpers.coffee]: This is just so we can more easily form the label later on and so we can show the time before live bidding starts (unless i'm misunderstanding your question :))

File: apps/home/client/setup_home_page_modules.coffee (hunk 3 above)
Comment (sweir27) [inline apps/home/client/setup_home_page_modules.coffee]: whoa coffeescript!

**Comment by sweir27** [comment]
@broskoski thanks for the comments! I'll merge this Monday morning so I can test it a whole bunch before deploying 😳

**Comment by joeyAghion** [comment]
I'd like to understand if this aims to handle missing `end_at` values more gracefully in general, or whether it imposes a new expectation that sales with missing `end_at` are of the "live" variety and have other values such as `live_start_at`.

My preference would be to handle the missing values more gracefully in general (such as by simply not presenting any countdown when no corresponding value is available). If that's difficult given the UI, I at least want to understand the new assumptions so we can enforce them consistently.

**Comment by sweir27** [comment]
This does not handle missing `end_at` in all cases, but only for the case where a `live_sale_at` exists. Totally agree that it would be best to handle a missing `end_at` in all cases, but this felt like a shorter term step in the right direction. I think there are some existing reasons why we want to require and `end_at` to be set for online-only sales, both because of the existing UI and the process around ending sales (which at this point _should_ be automatic in those cases).

That being said, I'm not against discussing the potential UI implications of a sale that has no `end_at` set with @katarinabatina and returning to this, since I can already see our conversation in 2 months when suddenly we don't require any of these fields 😅 .

**Pull request merged**